### PR TITLE
support datastore password secret created by external resources

### PR DESCRIPTION
**Pull request by grameshtwilio**
This allows the datastore secret object to be created by external agents such as https://github.com/external-secrets/external-secrets

### CODE REVIEW

**Comment by faisal-memon** [comment]
Thanks @grameshtwilio for this submission. I think this will be a useful feature.

File: charts/spire/charts/spire-server/README.md [hunk 1]
```diff
@@ -123,6 +123,9 @@ In order to run Tornjak with simple HTTP Connection only, make sure you don't cr
 | controllerManager.service.port | int | `443` |  |
 | controllerManager.service.type | string | `"ClusterIP"` |  |
 | controllerManager.validatingWebhookConfiguration.failurePolicy | string | `"Fail"` |  |
+| dataStore.dbSecret | object | `{"key":"DBPW","name":""}` | The secret should reside in the same namespace as the spire server |
+| dataStore.dbSecret.key | string | `"DBPW"` | The key of the secret object whose value is the dataStore password |
+| dataStore.dbSecret.name | string | `""` | The name of the secret object |
```
Comment (faisal-memon) [inline charts/spire/charts/spire-server/README.md:128]: What do you think of putting this under `datastore.sql.externalSecret` or something like that?  Would an enable flag help here so you don't have to check name is not empty to enable the feature?

**Comment by grameshtwilio** [comment]
Thanks @faisal-memon , I believe I have addressed your suggestions. Open to more suggestions and feedback!

**Review by marcofranssen — APPROVED** [review]
LGTM 🚀

**Review by faisal-memon — APPROVED** [review]
Looks good, thanks for making the requested changes.

**Pull request merged**